src/libraries/System.Reflection.TypeExtensions/tests/MetadataTokenTests.cs

```diff
@@ -46,14 +46,17 @@ public void SuccessImpliesNonNilWithCorrectTable(MemberInfo member, int expected
         }
 
         [ConditionalFact(nameof(GetMetadataTokenSupported), nameof(IsReflectionEmitSupported))]
-        public static void UnbakedReflectionEmitType_HasNoMetadataToken()
+        public static void ReflectionEmitType_HasMetadataToken()
         {
             AssemblyBuilder assembly = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("dynamic"), AssemblyBuilderAccess.Run);
             ModuleBuilder module = assembly.DefineDynamicModule("dynamic.dll");
             TypeBuilder type = module.DefineType("T");
             MethodInfo method = type.DefineMethod("M", MethodAttributes.Public);
-            Assert.False(method.HasMetadataToken());
-            Assert.Throws<InvalidOperationException>(() => method.GetMetadataToken());
+
+            // Previously get_MetadataToken was always thrown due to MemberInfo.get_MetadataToken not being
+            // overridden in MethodBuilder. Now MethodBuilder overrides that and creates the token.
```

Comment:
```suggestion
```
Nit: We typically do not add comments in tests about what the previous behavior was. You can always find it in the history if needed.